metrics/cgroups/v2/cpu.go

```diff
@@ -0,0 +1,124 @@
+// +build linux
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package v2
+
+import (
+	v2 "github.com/containerd/containerd/metrics/types/v2"
+	metrics "github.com/docker/go-metrics"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var cpuMetrics = []*metric{
+	{
+		name: "cpu_usage",
+		help: "Total cpu usage (cgroup v2)",
+		unit: metrics.Unit("microseconds"),
+		vt:   prometheus.GaugeValue,
+		getValues: func(stats *v2.Metrics) []value {
+			if stats.CPU == nil {
+				return nil
+			}
+			return []value{
+				{
+					v: float64(stats.CPU.UsageUsec),
+				},
+			}
+		},
+	},
+	{
+		name: "cpu_user_usage",
+		help: "Current cpu usage in user space (cgroup v2)",
+		unit: metrics.Unit("microseconds"),
+		vt:   prometheus.GaugeValue,
+		getValues: func(stats *v2.Metrics) []value {
+			if stats.CPU == nil {
+				return nil
+			}
+			return []value{
+				{
+					v: float64(stats.CPU.UserUsec),
+				},
+			}
+		},
+	},
+	{
+		name: "cpu_system_usage",
```

Comment:
Still inconsistent. This should be either `cpu_system_usec` (v2 convention) or `cpu_kernel` (v1-compatible convention, if the semantic is same as v1 `cpu_kernel`. I haven't checked whether these values are compatible enough.)

Reply:
Same applies to other CPU fields and `ctr` output.

Reply:
```
cpuacct.stat
reports the user and system CPU time consumed by all tasks in this cgroup (including tasks lower in the hierarchy) in the following way:
user — CPU time consumed by tasks in user mode.
system — CPU time consumed by tasks in system (kernel) mode.
```
Even in cgroupv1 `system` name was used, `kernel` used just for more transparency.
More than that in v1 metrics `cpu_kernel` name is used, which one I don't like at all.
So, we can use `cpu_system_usage` name to make it closer to original documentation, or `cpu_kernel_usage` to make it more clear.

Reply:
why not `cpu_system_usec`?

Reply:
Since we already pointed metrics units bellow(milliseconds)

Reply:
I think we should prioritize consistency over human friendliness.

Reply:
Agree, but the question is, consistency with what? with v1 or with v2 official documentation?
Since v1 metrics in containerd not consistent with v1 docs.

Reply:
Generally this should be consistent with v2 official.
If the meaning is exactly same, the field name can be consistent with containerd's v1 implementation for preserving compatibility, as discussed in https://github.com/containerd/containerd/issues/3726#issuecomment-565155925

Reply:
At least units are not the same, v1 - nanoseconds, v2 - milliseconds.
So i think we can take official v2 naming `cpu_system_usec` and `usage_usec, user_usec` correspondingly.

Reply:
:+1: for `cpu_system_usec`

Reply:
What about to rename `cpu_usage` to `cpu_usage_usec` and `cpu_user_usage` to `cpu_user_usec` to make it completely consistent with docs?

Reply:
yes, please